uint128/uint128.go

```diff
@@ -0,0 +1,122 @@
+package uint128
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+type Int [2]uint64
+
+func SetUint64(lo uint64) *Int {
+	// i[1] = high bits
+	// i[0] = low bits
+	z := &Int{}
+	z.setUint128(0, lo)
+	return z
+}
+func SetUint128(hi, lo uint64) *Int {
+	z := &Int{}
+	z.setUint128(hi, lo)
+	return z
+}
+
+func SetInt128(hex_str string) *Int {
+	z := &Int{}
+
+	z.parseHexString(hex_str)
+
+	return z
+}
+
+func (i *Int) setUint128(hi, lo uint64) error {
+	loBytes := make([]byte, 8)
+	hiBytes := make([]byte, 8)
+
+	binary.BigEndian.PutUint64(loBytes, lo)
+	binary.BigEndian.PutUint64(hiBytes, hi)
+
+	err := i.setBytes(append(hiBytes, loBytes...))
+	if err != nil {
+		return fmt.Errorf("couldn't marshal %016x%016x into a uint128 type", lo, hi)
+	}
+
+	return nil
+}
+
+func (i *Int) Bytes() []byte {
+	hiBytes := make([]byte, 8)
+	loBytes := make([]byte, 8)
+
+	binary.BigEndian.PutUint64(hiBytes, i[1])
+	binary.BigEndian.PutUint64(loBytes, i[0])
+
+	return append(hiBytes, loBytes...)
+}
```

Comment:
```suggestion
func (i *Int) Bytes() []byte {
	result := make([]byte, 16)
	binary.BigEndian.PutUint64(result[8:16], i[0])
	binary.BigEndian.PutUint64(result[0:8], i[1])
	return result
}
```
We can save some allocations this way.